pkg/tfgen/examples_coverage_exporter.go

```diff
@@ -0,0 +1,158 @@
+}
+
+// The second mode, similar to existing Pulumi coverage Json files uploadable to redshift
+func (CE *CoverageExportUtil) exportUploadableResults(outputDirectory string, fileName string) {
+
+	// The Coverage Tracker data structure is flattened down to the example level, and they all
+	// get individually written to the file in order to not have the "{ }" brackets at the start and end
+	type SingleExampleResult struct {
+		ProviderName    string
+		ProviderVersion string
+		ExampleName     string
+		FailedLanguages []LanguageConversionResult
+	}
+
+	jsonOutputLocation := createJsonOutputLocation(outputDirectory, fileName)
+
+	// All the examples in the map are iterated by key and marshalled into one large byte array
+	// separated by \n, making the end result look like a bunch of Json files that got concatenated
+	var result = []byte{}
+	for _, exampleInMap := range CE.CT.EncounteredExamples {
+		singleExample := SingleExampleResult{
+			ProviderName:    CE.CT.ProviderName,
+			ProviderVersion: CE.CT.ProviderVersion,
+			ExampleName:     exampleInMap.Name,
+			FailedLanguages: []LanguageConversionResult{},
+		}
+
+		for _, conversionResult := range exampleInMap.LanguagesConvertedTo {
+			if conversionResult.FailureSeverity > 0 {
+				singleExample.FailedLanguages = append(singleExample.FailedLanguages, conversionResult)
+			}
+		}
+		marshalledExample, err := json.MarshalIndent(singleExample, "", "\t")
+		panicIfError(err, "Failed to MarshalIndent JSON file")
+		result = append(append(result, marshalledExample...), uint8('\n'))
+	}
+	err2 := ioutil.WriteFile(jsonOutputLocation, result, 0600)
+	panicIfError(err2, "Failed to write JSON file")
+}
+
+// The third mode, meant for exporting broad information such as total number of examples,
+// and what percentage of the total each failure severity makes up
+func (CE *CoverageExportUtil) exportSummarizedResults(outputDirectory string, fileName string) {
+
+	// The Coverage Tracker data structure is used to gather general statistics about the examples
+	type NumPct struct {
+		Number int
+		Pct    float32
+	}
+
+	type CoverageStatistics struct {
+		NoErrors      NumPct
+		LowSevErrors  NumPct
+		HighSevErrors NumPct
+		Fatal         NumPct
+		Total         int
+	}
+
+	SummarizedResult := CoverageStatistics{NumPct{0, 0.0}, NumPct{0, 0.0}, NumPct{0, 0.0}, NumPct{0, 0.0}, 0}
+
+	//All the language conversion attempts in each example are iterated by key and analyzed
+	for _, exampleInMap := range CE.CT.EncounteredExamples {
+		for _, conversionResult := range exampleInMap.LanguagesConvertedTo {
+			SummarizedResult.Total += 1
+			if conversionResult.FailureSeverity == 0 {
+				SummarizedResult.NoErrors.Number += 1
+			} else if conversionResult.FailureSeverity == 1 {
+				SummarizedResult.LowSevErrors.Number += 1
+			} else if conversionResult.FailureSeverity == 2 {
+				SummarizedResult.HighSevErrors.Number += 1
+			} else {
+				SummarizedResult.Fatal.Number += 1
+			}
```

Comment:
This is well-suited for a `switch` statement:

```suggestion
			switch conversionResult.FailureSeverity {
			case 0:
				SummarizedResult.NoErrors.Number++
			case 1:
				SummarizedResult.LowSevErrors.Number++
			case 2:
				SummarizedResult.HighSevErrors.Number++
			default:
				SummarizedResult.Fatal.Number += 1
			}
```

Reply:
Also, it looks like there are `const`s for these numbers defined elsewhere, so please use those rather than the numbers in these cases.